### PR TITLE
Add missing metadata to MODIS L1b and L2 readers

### PR DESCRIPTION
**Pull request by djhoese**
I noticed that the angle datasets from the MODIS readers were missing the sensor and platform_name metadata. I also needed to add `rows_per_scan` to match what I was doing in the old Polar2Grid reader. Also turns out there were no tests for the 'modis_l1b' reader so those were added as well. :vomiting_face: 

 - [x] Tests added

### CODE REVIEW

File: satpy/tests/reader_tests/test_modis_l1b.py
Comment (mraspaud) [inline satpy/tests/reader_tests/test_modis_l1b.py:3]:
```suggestion
# Copyright (c) 2019-2021 Satpy developers
```

File: satpy/tests/reader_tests/test_modis_l1b.py
Comment (mraspaud) [inline satpy/tests/reader_tests/test_modis_l1b.py:231]: This class should be moved to a dedicated test file or alternatively this file should be renamed.

**Review by mraspaud — COMMENTED** [review]
Looks good, just a couple of small comments

**Comment by djhoese** [comment]
This also updates the `modis_l1b.yaml` so that it doesn't incorrectly state that 250m and 500m versions of the angle datasets can be loaded from the 1000m file. It is currently not possible to interpolate the 5000m data to a higher resolution than 1000m.

File: satpy/tests/reader_tests/test_modis_l1b.py
Comment (djhoese) [reply satpy/tests/reader_tests/test_modis_l1b.py:3]: This should probably be just 2021. This file is new.

File: satpy/tests/reader_tests/test_modis_l1b.py
Comment (djhoese) [reply satpy/tests/reader_tests/test_modis_l1b.py:236]: This is actually just misnamed since it was copy/pasted from the L2 tests. It should be TestModisL1b.

**Comment by djhoese** [comment]
Assuming coveralls isn't dead today this should run and succeed. Once done I'll merge it.

I acknowledge that there is shared code between the modis_l1b and modis_l2 test modules, but I'd like to wait until these are merged and I work more on #1815 to refactor and make things better. I'll have a better understanding of what files are needed and how each file should be represented.

**Review by mraspaud — APPROVED** [review]
LGTM!

**Pull request merged**
